unit-tests/py/rspy/acroname.py

```diff
@@ -58,38 +59,71 @@ def discover():
     # see https://acroname.com/reference/_modules/brainstem/module.html#Module.discoverAndConnect
     try:
         log.debug_indent()
-        specs = brainstem.discover.findAllModules( brainstem.link.Spec.USB )
+        for i in range(retries + 1):
+            specs = brainstem.discover.findAllModules( brainstem.link.Spec.USB )
+            if not specs:
+                time.sleep(1)
+            else:
+                for spec in specs:
+                    log.d( '...', spec )
+                break
         if not specs:
             raise NoneFoundError()
-        for spec in specs:
-            log.d( '...', spec )
     finally:
         log.debug_unindent()
 
     return specs
 
 
-def connect( spec = None ):
+def connect( reset = False, req_spec = None ):
     """
     Connect to the hub. Raises RuntimeError on failure
+    :param reset: When true, the acroname will be reset as part of the connection process
+    :param req_spec: Required spec to connect to.
     """
 
     global hub
     if not hub:
         hub = brainstem.stem.USBHub3p()
 
-    if spec:
-        specs = [spec]
+    if req_spec:
+        spec = req_spec
     else:
         specs = discover()
         spec = specs[0]
 
     result = hub.connectFromSpec( spec )
     if result != brainstem.result.Result.NO_ERROR:
-        raise RuntimeError( "failed to connect to acroname (result={})".format( result ))
+        raise RuntimeError( "failed to connect to Acroname (result={})".format( result ))
     elif len(specs) > 1:
```

Comment:
`specs` will not always exist, e.g. if `req_spec` is not `None`

Reply:
Will fix